workspaces/ui-v2/src/pages/docs/components/DeleteEndpointConfirmationModal.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { FC } from 'react';
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  makeStyles,
+} from '@material-ui/core';
+
+import { EndpointName } from '<src>/components';
+import { IEndpoint } from '<src>/types';
+
+type DeleteEndpointConfirmationModalProps = {
+  endpoint: IEndpoint;
+  handleClose: () => void;
+  handleConfirm: () => void;
+};
+
+export const DeleteEndpointConfirmationModal: FC<DeleteEndpointConfirmationModalProps> = ({
+  endpoint,
+  handleClose,
+  handleConfirm,
+}) => {
+  const classes = useStyles();
+  return (
+    <Dialog open={true} onClose={handleClose}>
+      <DialogTitle>Confirm delete endpoint</DialogTitle>
+      <DialogContent>
+        <DialogContentText>
+          <div className={classes.confirmationTextContainer}>
+            Are you sure you want to delete endpoint the following endpoint?
+            <EndpointName
+              method={endpoint.method}
+              fullPath={endpoint.fullPath}
+            />
+          </div>
+          <div className={classes.confirmationTextContainer}>
+            This will permanently delete this endpoint. You will lose all
+            contributions associated with this endpoint and will no longer see
+            this endpoint in your documentation page.
```

Comment:
Is there anything else we should raise to the user (with how we plan to implement delete in the domain model)?

Reply:
Once we've also implemented revert, this wouldn't have as much of a disastrous impact, as the delete could be reverted. Also, we do actually still store the contributions (indefinitely until we implement implement compaction of the specification.json). 

That said, better warn sternly here and have it turn out not that bad.